node/bundler.ts

```diff
@@ -32,11 +32,11 @@ interface BundleOptions {
   onAfterDownload?: OnAfterDownloadHook
   onBeforeDownload?: OnBeforeDownloadHook
   systemLogger?: LogFunction
-  internalSrcFolder?: string
 }
 
 const bundle = async (
   sourceDirectories: string[],
+  internalSourceDirectory: string,
```

Comment:
I'm not sure why you're moving this parameter and introducing a breaking change in the process. Is it because you want to make it mandatory? If so, why? If we don't receive an internal source directory, we can just consider all functions to be user-generated? Or am I missing something?

Reply:
Yeah the API is not perfect. the problem i had conceptually was the before this PR we supplied the user and internal source directory in `sourceDirectories` and additionally supplied the same internal path in the options. In order to split the configs we need to know which path from `sourceDirectories` is the internal one.

What I tried by changing the API is making it somehow clear that when one supplies an internal function directory in `sourceDirectories` that one also have to specify the optional `internalSourcePath` option, as otherwise the generated result might be wrong.

So instead I opted for removing the internalSourcePath `option` removing it from `sourceDirectories` and adding it as its own parameter, because `build` sets it anyway always.

I could move the param back to options and make it optional, but I find this a weird API that one of the main inputs is hidden in the options. `edge-bundler` API doesn't seem to be currently built for the use case of differentiating between directories that have a different type.

One other option I thought about was to supply `sourceDirectories` not as `string[]` but instead `{ dir: string, type: "internal"| "user" }[]` or similar.

Not sure what would be better.

Reply:
I see what you mean. For what it's worth, we use that [same concept in zip-it-and-ship-it](https://github.com/netlify/zip-it-and-ship-it#internalsrcfolder). The initial idea was to keep these packages as dumb as possible, but that's no longer the case as Edge Bundler needs to know how to order the declarations.

I still think the current API works as long as we use it properly, and keeping it saves you the hassle of handling breaking changes.

But I'm happy to defer to you. Just wanted to clarify this and ensure you were aware that the data is already being passed in.